sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala

```diff
@@ -79,6 +78,23 @@ private[state] class HDFSBackedStateStoreProvider extends StateStoreProvider wit
   //   java.util.ConcurrentModificationException
   type MapType = java.util.concurrent.ConcurrentHashMap[UnsafeRow, UnsafeRow]
 
+  class HDFSBackedReadOnlyStateStore(val version: Long, map: MapType)
+    extends ReadOnlyStateStore {
+
+    override def id: StateStoreId = HDFSBackedStateStoreProvider.this.stateStoreId
+
+    override def get(key: UnsafeRow): UnsafeRow = map.get(key)
+
+    override def abort(): Unit = {}
```

Comment:
Isn't this a `ReadOnlyStateStore`? Why need to override `abort`? Can't we have `throwNotAllowed` too for `abort` in `ReadOnlyStateStore`?

Reply:
abort() could be called for ReadOnlyStateStore as well, as the code path is not diverged between ReadOnlyStateStore/StateStore to minimize the code diff. But looks like it may bring more confusion - probably better to separate the code path clearly despite of more code changes. I'll look into it again.